pkgs/development/compilers/factor-lang/default.nix

```diff
@@ -0,0 +1,103 @@
+{ stdenv, fetchurl, fetchFromGitHub, glib, glibc, git,
+  rlwrap, curl, pkgconfig, perl, makeWrapper, tzdata, ncurses,
+  libX11, pango, cairo, gtk2, gdk_pixbuf, gtkglext,
+  mesa, libXmu, libXt, libICE, libSM }:
+
+stdenv.mkDerivation rec {
+  name = "factor-lang-${version}";
+  version = "0.97";
+  rev = "eb3ca179740e6cfba696b55a999caa13369e6182";
+
+  src = fetchFromGitHub {
+    owner = "factor";
+    repo = "factor";
+    rev = rev;
+    sha256 = "16zlbxbad3d19jq01nk824i19bypqzn8l3yfxys40z06vjjncapd";
+  };
+
+  factorimage = fetchurl {
+    url = http://downloads.factorcode.org/releases/0.97/factor-linux-x86-64-0.97.tar.gz;
+    sha256 = "06y125c8vbng54my5fxdr3crpxkvhhcng2n35cxddd3wcg6vhxhp";
+    name = "factorimage";
+  };
+
+  buildInputs = [ git rlwrap curl pkgconfig perl makeWrapper
+    libX11 pango cairo gtk2 gdk_pixbuf gtkglext
+    mesa libXmu libXt libICE libSM ];
+
+  buildPhase = ''
+    make $(bash ./build-support/factor.sh make-target) GIT_LABEL=heads/master-${rev}
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    # First, get a workable image. Unfortunately, no boot-image
+    # is available with release info. So fetch a released image.
+    # The released image has library path info embedded, so we
+    # have to first recreate the boot image with Nix paths, and
+    # then use it to build the Nix release image.
+    (cd $out/ &&
+      zcat ${factorimage} | tar -xvpf - factor/factor.image &&
+      mv factor/factor.image bin/.factor-wrapped.image &&
+      rm -rf factor )
+
+    cp -r basis core extra unmaintained $out/bin
```

Comment:
Unfortunately, factor looks for the libraries and runtime image in the directory where the binary was found.